datalake/src/main/java/com/sequenceiq/datalake/settings/SdxRepairSettings.java

```diff
@@ -0,0 +1,36 @@
+package com.sequenceiq.datalake.settings;
+
+import java.util.List;
+
+import javax.ws.rs.BadRequestException;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.util.Strings;
+
+import com.sequenceiq.sdx.api.model.SdxRepairRequest;
+
+public class SdxRepairSettings {
+
+    private List<String> hostGroupNames;
+
+    private SdxRepairSettings() {
+    }
+
+    public static SdxRepairSettings from(SdxRepairRequest request) {
+        if (StringUtils.isNotBlank(request.getHostGroupName()) && CollectionUtils.isNotEmpty(request.getHostGroupNames())) {
+            throw new BadRequestException("Please send only one hostGroupName in the 'hostGroupName' field " +
+                    "or multiple hostGroups in the 'hostGroupNames' fields");
+        }
+        SdxRepairSettings settings = new SdxRepairSettings();
+        if (Strings.isNotEmpty(request.getHostGroupName())) {
+            settings.hostGroupNames = List.of(request.getHostGroupName());
+        }
+        settings.hostGroupNames = request.getHostGroupNames();
```

Comment:
I think this statement shall be in the `else` clause of the `if` above, otherwise `settings.hostGroupNames` might get overwritten (with `null` or whatever is passed in `request.getHostGroupNames()`) even if a single host group was to be repaired. Alternatively, the statement can be also moved before the `if`, whichever approach you prefer. 😄